robotnealan/bikesy-tahoe
# Create appConfig for regional configuration

## Overview
Enable regional configuration in the same repo.

The way it works is the developer sets their env var like `NEXT_PUBLIC_REGION=sf`, and that selects to include a whole bunch of different region-specific configuration values, specified in `sf.js`. This way non-secret configuration values can be committed to source code. And non-region-specific configuration values can also be shared in the default.js.

The motivation for doing this is to avoid maintaining a large and growing `.env` file outside of source control. Multiple developers working on the same project (like me and Rob in Tahoe) would need to share this file with each other outside of git, and in some reasonably secure way because it contains secrets. Also it would need to stay in sync with the individual configuration values in the Heroku web UI. Also, as I continue to add more and more region-specific configuration, I can update the values for all regions in one PR, instead of having to ask the maintainers of each instance of Bikesy to update their local environment files.


## Suggested environment changes to accompany this PR

- Different instantiations of bikesy are expected to set a new environment variable `NEXT_PUBLIC_REGION` to configure which regional config file it should use. If unset, I set the default value of this to 'sf', so that no changes should be needed to the Bikesy.com heroku environment in order to deploy this change.
- But I would still recommend updating the environment, after deploy, to bring it inline with the suggestions in .env.example: setting `NEXT_PUBLIC_REGION` to `sf`, and removing all the other environment configurations that are not secrets.
- I recommend doing the same thing for your local gitignored `.env` file too.

## Testing:

- Searched the repo for `process.env` and ensured that all remaining instances are expected secrets.
- Tried querying for directions. Searching an address within my search bounds "worked" (no error message). Searching an address outside my search bounds yielded an error saying that the address is out of bounds.
- Confirmed that with `CONFIG=tahoe` and `NEXT_PUBLIC_MAPBOX_ACCESS_TOKEN=<TahoeBike access token>`, the map loads visually correctly. 
- Confirmed that with `CONFIG=sf` and `NEXT_PUBLIC_MAPBOX_ACCESS_TOKEN=<TahoeBike access token>`, the map fails to load with a 404 from Mapbox in the console. Makes sense. I can't load bikesy styles with my access token. (And I don't have the bikesy access token)
- Toggled the values of WELCOME_MODAL_TITLE and SHOULD_SHOW_WELCOME_MODAL and confirmed the expected behavior.
- Make sure that routing generally works as expected
- Remove NEXT_PUBLIC_REGION configuration and confirm things generally look right for SF.
